models/models.js

```diff
@@ -39,13 +39,42 @@ exports.selectAllUsers = async () => {
   return rows;
 };
 
-exports.selectArticles = async () => {
-  const { rows } = await db.query(
-    `SELECT CAST(COUNT(c.comment_id)as int) as comment_count, a.author,a.title,a.article_id, a.topic , a.created_at, a.votes 
-  FROM articles AS a
-  LEFT  JOIN comments as c ON c.article_id = a.article_id
-  GROUP BY a.author,a.title,a.article_id, a.topic , a.created_at, a.votes
-  ORDER BY a.created_at DESC;`
-  );
-  return rows;
+exports.selectArticles = async (
+  sort_by = "created_at",
+  order = "DESC",
+  topic = "*"
+) => {
+  const niceSortBys = [
+    "article_id",
+    "author",
+    "topic",
+    "comment_count",
+    "created_at",
+    "votes",
+    "title",
+  ];
+  const niceOrder = ["DESC", "ASC"];
+  const niceTopics = ["mitch", "cats", "paper", "*"]; ///will have to be changed as topic added
```

Comment:
You could use your allTopics endpoint here so it doesn't have to be updated as people add topics. If you use Reddit at all - people can add a new page on a subject whenever they want... in real life that would be a nightmare to handle manually.